table.go

```diff
@@ -255,26 +354,55 @@ func (t *Table) InsertBuffer(ctx context.Context, buf *dynparquet.Buffer) (uint6
 }
 
 func (t *Table) Insert(ctx context.Context, buf []byte) (uint64, error) {
+	return t.insert(ctx, buf)
+}
+
+func (t *Table) appendToLog(ctx context.Context, tx uint64, buf []byte) error {
+	if err := t.wal.Log(tx, &walpb.Record{
+		Entry: &walpb.Entry{
+			EntryType: &walpb.Entry_Write_{
+				Write: &walpb.Entry_Write{
+					Data:      buf,
+					TableName: t.name,
+				},
+			},
+		},
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (t *Table) insert(ctx context.Context, buf []byte) (uint64, error) {
+start:
+	// Using active write block is important because it ensures that we don't
+	// miss pending writers when synchronizing the block.
+	block, close := t.ActiveWriteBlock()
+	if block.Size() > t.db.columnStore.activeMemorySize {
+		if err := t.RotateBlock(block); err != nil {
+			close()
+			return 0, fmt.Errorf("rotate block: %w", err)
+		}
+		close()
+		goto start
```

Comment:
I'd like to refactor this out, gotos create spaghetti

Reply:
After playing with this code a bit, I think we can actually do this without the goto.